File: update-questions.py
import json
import re

# Read extracted questions
with open('extracted_all_questions.json', 'r', encoding='utf-8') as f:
    extracted = json.load(f)

# Read questions-data.js
with open('questions-data.js', 'r', encoding='utf-8') as f:
    content = f.read()

def escape_quotes(text):
    return text.replace("'", "\\'")

# Update each year's questions
for year, questions in extracted.items():
    if year == 'metadata':
        continue

    print(f"Processing year {year}...")

    for qid, qdata in questions.items():
        if not qdata.get('questionTextDE'):
            continue

        # Find and replace question text
        pattern = rf"(id: '{qid}',\s+section: '[ABC]',\s+number: \d+,\s+year: {year},\s+points: [345],\s+)questionTextDE: '[^']*',\s+questionTextEN: '[^']*',"

        replacement = (
            r"\1"
            f"questionTextDE: '{escape_quotes(qdata['questionTextDE'])}',\n"
            f"                questionTextEN: '{escape_quotes(qdata['questionTextEN'])}',"
        )

        content = re.sub(pattern, replacement, content)

        # Update options if they have text
        if qdata.get('hasTextOptions') and qdata.get('options'):
            options_str = "[\n"
            for opt in qdata['options']:
                options_str += f"                    {{ id: '{opt['id']}', textDE: '{escape_quotes(opt['textDE'])}', textEN: '{escape_quotes(opt['textEN'])}' }},\n"
            options_str += "                ]"

            # Replace options array
            opt_pattern = rf"(year: {year},.*?id: '{qid}'.*?options: )\[[^\]]*\]"
            content = re.sub(opt_pattern, r"\1" + options_str, content, flags=re.DOTALL)

# Write back
with open('questions-data.js', 'w', encoding='utf-8') as f:
    f.write(content)

print("✓ Questions data updated successfully!")

File: update-questions.js
// Script to merge extracted questions into questions-data.js
const fs = require('fs');

// Load extracted questions
const extracted = JSON.parse(fs.readFileSync('extracted_all_questions.json', 'utf8'));

// Load existing questions data
const questionsDataPath = 'questions-data.js';
let questionsFileContent = fs.readFileSync(questionsDataPath, 'utf8');

// Function to update a question in the data
function updateQuestion(year, questionId, extractedQ) {
    const searchPattern = new RegExp(
        `(\\s+{\\s+id: '${questionId}',\\s+section: '[ABC]',\\s+number: \\d+,\\s+year: ${year},\\s+points: [345],\\s+)` +
        `questionTextDE: '[^']*',\\s+questionTextEN: '[^']*',([^}]*?options: \\[[^\\]]*\\])`,
        'gs'
    );

    // Build replacement
    let optionsStr = '';
    if (extractedQ.hasTextOptions && extractedQ.options) {
        const opts = extractedQ.options.map(opt =>
            `{ id: '${opt.id}', textDE: '${opt.textDE.replace(/'/g, "\\'")}', textEN: '${opt.textEN.replace(/'/g, "\\'")}' }`
        ).join(',\n                    ');
        optionsStr = `\n                options: [\n                    ${opts}\n                ],`;
    }

    const replacement = `$1questionTextDE: '${extractedQ.questionTextDE.replace(/'/g, "\\'")}',\n                questionTextEN: '${extractedQ.questionTextEN.replace(/'/g, "\\'")}',${optionsStr}`;

    questionsFileContent = questionsFileContent.replace(searchPattern, replacement);
}

// Update questions for each year
for (const [year, questions] of Object.entries(extracted)) {
    if (year === 'metadata') continue;

    console.log(`Updating year ${year}...`);
    for (const [questionId, questionData] of Object.entries(questions)) {
        if (questionData.questionTextDE) {
            updateQuestion(year, questionId, questionData);
        }
    }
}

// Save updated file
fs.writeFileSync(questionsDataPath, questionsFileContent, 'utf8');
console.log('Questions data updated successfully!');
